src/example-types/results.js

```diff
@@ -28,15 +28,32 @@ module.exports = {
     if (context.include || context.exclude) result._source = {}
     if (context.include) result._source.includes = context.include
     if (context.exclude) result._source.excludes = context.exclude
+
     let highlight =
       _.getOr(true, 'highlight', context) && schema.elasticsearch.highlight
+    let inlineAliases = _.getOr(
+      {},
+      'elasticsearch.highlight.inlineAliases',
+      schema
+    )
+
     if (highlight) {
-      let schemaHiglightFields = _.flatten(
-        _.values(schema.elasticsearch.highlight)
-      )
-      let highlightFields = _.get('showOtherMatches', context)
-        ? schemaHiglightFields
-        : _.intersection(context.include, schemaHiglightFields)
+      // Only take the fields that matter to highlighting which are the inline, inlineAliases and additionalFields sections
+      let schemaHiglightFields = _.flow(
+        _.pick(['inline', 'additionalFields']),
+        _.values,
+        _.flatten,
+        _.concat(_.keys(inlineAliases)),
```

Comment:
Is this to catch the case where a field is in `inlineAlias` but _not_ `inline`?

Reply:
Well yes we need to combine all fields that are related to highlighting and then via the intersection only take those which are relevant in the context.